### ISSUE TITLE
torch.distributions bug in RelaxedOneHotCategorical.log_prob 

### ISSUE THREAD

**Issue by kdharris101** [pytorch/pytorch#37162]
## 🐛 Bug

I'm a new user so maybe I misunderstood something - but I think I found a bug in torch.distributions.RelaxedOneHotCategorical. I also think I have a fix for it.

I would like to create a batch of n distributions with different temerature parameters.  I believe the way to do this is to pass in temperature of size [n,1], and logits of size [n,k] (where k is the number of categories). However, calling log_prob returns an output of size [n,n] rather than [n,1] or just [n]

I believe the fix is to change line 83 of relaxed_categorical.py to:

`
score = (score - score.logsumexp(dim=-1, keepdim=True)).sum(-1, keepdim=True)
`

Also: to access ExpRelaxedCategorical you have to say td.relaxed_categorical.ExpRelaxedCategorical, rather than just td.ExpRelaxedCategorical. Can this be fixed too?

## To Reproduce

```
dist = td.relaxed_categorical.ExpRelaxedCategorical(
    temperature=torch.ones(5,1), logits = torch.zeros(5,2)
)
print(dist.log_prob(torch.ones(5,2)/2).shape)
````

Output is:
```
torch.Size([5, 5])
```

I thought it should be torch.Size([5,1]) or torch.Size([5]

## Environment
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.8
Is CUDA available: No
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GPU
Nvidia driver version: 391.40
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] Could not collect
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h74a9793_0  
[conda] mkl                       2020.0                      166  
[conda] mkl-service               2.3.0            py38hb782905_0  
[conda] mkl_fft                   1.0.15           py38h14836fe_0  
[conda] mkl_random                1.1.0            py38hf9181ef_0  
[conda] numpy                     1.18.1           py38h93ca92e_0  
[conda] numpy-base                1.18.1           py38hc3f5095_1  
[conda] pytorch                   1.4.0           py3.8_cuda101_cudnn7_0    pytorch
[conda] torchvision               0.5.0                py38_cu101    pytorch

**Comment by mctigger** [comment]
The bug still exists to this day.
